lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -109,14 +110,23 @@ def test_not_a_file(self, query_get_as, monkeypatch):
 
     def test_dataset_in_given_path(self, query_get_as, monkeypatch):
         mock_args: Optional[tuple[Path, dict[str, Any]]] = None
+        exp_stream = io.BytesIO(b"file_as_a_byte_stream")
+        filestream_dict = {
+            "name": "f1.json",
+            "type": CacheType.FILE,
+            "stream": exp_stream,
+        }
 
         def mock_send_file(path_or_file, *args, **kwargs):
             nonlocal mock_args
+            assert path_or_file == exp_stream
             mock_args = (path_or_file, kwargs)
             return {"status": "OK"}
 
         monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
-        monkeypatch.setattr(Path, "is_file", lambda self: True)
+        monkeypatch.setattr(
+            self.MockTarball, "filestream", lambda _s, _t: filestream_dict
+        )
         monkeypatch.setattr(
             "pbench.server.api.resources.datasets_inventory.send_file", mock_send_file
         )
```

Comment:
There's no mock for `close()`...should there be?  (Given that you have `exp_stream`, I think you can patch it directly to catch _its_ `close()` method.)